kms/tpmkms/tpmkms_simulator_test.go

```diff
@@ -59,10 +59,19 @@ func withKey(name string) newSimulatedTPMOption {
 	}
 }
 
-func newSimulatedTPM(t *testing.T, opts ...newSimulatedTPMOption) *tpmp.TPM {
+func newSimulatedTPM(t *testing.T, caps *tpmp.Capabilities, opts ...newSimulatedTPMOption) *tpmp.TPM {
```

Comment:
Can you turn the capabilities into a `newSimulatedTPMOption`, like `withAK` and `withKey`, only not operating on the `tpm.TPM` instance? Then it can remain variadic, and won't require changing it in the tests where you're now passing `nil`. It'll still need the special case to properly create the TPM instance, so there's still some logic needed to switch between "initialization" and "preparation" options.

Reply:
Yea, I'll work on this more - I wanted to do that originally, but I think I'll have to change how options work between this package and the `tpm` package to be able to actually modify the capabilities.

Reply:
I think it can be done by changing the `newSimulatedTPMOption` to also return a type or bool (or to rework that into a struct), and to use that to differentiate between initialization and preparation options.

Reply:
@hslatman  Unless I'm not following you, I can't change this behavior without either exporting _something_ from the `tpm` package to control the capabilities while under test (eg: either a helper `Option` setter, the `options` struct and field, etc.). I've removed that test code for now: what do you think about either mocking the `tpm` package itself or adding support to the `Simulator` to override the capabilities, eg (in the `simulator` package):

```
// the packged command buffer for CmdGetCapability (0x17A)
var getCapabilityAlgs = []byte{128, 1, 0, 0, 0, 22, 0, 0, 1, 122, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1}

func (s *WrappingSimulator) Write(p []byte) (int, error) {
	if bytes.Equal(p, getCapabilityAlgs) {
		// ...
	}
	return s.wrapped.Write(p)
}
```

Reply:
Also, if you don't mind, I'll keep working on better tests for this after merging, so we can unblock work depending on these changes.

Reply:
I've added the test you had back in https://github.com/smallstep/crypto/pull/590 and merged that into this branch. You're right we'll need that `WithCapabilities` method exported; didn't think about the package boundary. I've marked it experimental, and noted that it's primarily used for testing. 

Some form of emulating responses from a TPM might be useful for future tests, and it would be great to have that available in plain Go, but I think it would be best in a new package, and not (partially) reliant on the C simulator. I know Panos has looked a bit into generating Go code from the spec, so that might be a way to do it.